### PR TITLE
Move to Apache 2 License and IBM copyright headers

### PR DESCRIPTION
**Pull request by guptakshit**
All files except Jenkinsfile are Terracotta Public License

### CODE REVIEW

File: pom.xml
Comment (guptakshit) [inline pom.xml:119]: I referred https://github.com/Terracotta-OSS/offheap-store/pull/83/commits/4e702534a149b6f39ecdaee585383d20c6e9c0f5

**Review by chrisdennis — CHANGES_REQUESTED** [review]
I'm going to say we should go ahead and make the TPL to Apache transition in this PR. Then let's leave it open pending the TPL decision later this week. Once made we can then just merge the one PR and be done.

**Comment by aurbroszniowski** [comment]
> I'm going to say we should go ahead and make the TPL to Apache transition in this PR. Then let's leave it open pending the TPL decision later this week. Once made we can then just merge the one PR and be done.

@chrisdennis 
If we need to wait for a decision for the TPL change, shouldn't we wait to make the changes in the code too? I'm asking because I have 2 other PRs with a fairly high number of TPL licenses so I'm wondering if I should already go ahead with changes to Apache, but if it's not sure, I don't want to make changes now and revert them later

**Comment by chrisdennis** [comment]
> > I'm going to say we should go ahead and make the TPL to Apache transition in this PR. Then let's leave it open pending the TPL decision later this week. Once made we can then just merge the one PR and be done.
> 
> @chrisdennis If we need to wait for a decision for the TPL change, shouldn't we wait to make the changes in the code too? I'm asking because I have 2 other PRs with a fairly high number of TPL licenses so I'm wondering if I should already go ahead with changes to Apache, but if it's not sure, I don't want to make changes now and revert them later

@aurbroszniowski we can wait sure... hopefully should know by end of Thursday at the latest.

**Comment by guptakshit** [comment]
following link https://github.com/Terracotta-OSS/bigmemory-core-storage/blob/1fc49360098658b75ed4b3457de774b39d8888db/Jenkinsfile#L18 needs to be replaced?

**Review by mathieucarbou — APPROVED** [review]

**Pull request merged**
